alerting/alerter-war/src/main/java/org/hawkular/metrics/alerter/ConditionExpression.java

```diff
@@ -0,0 +1,657 @@
+/*
+ * Copyright 2014-2017 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.metrics.alerter;
+
+import static org.hawkular.metrics.model.MetricType.AVAILABILITY;
+import static org.hawkular.metrics.model.MetricType.COUNTER;
+import static org.hawkular.metrics.model.MetricType.COUNTER_RATE;
+import static org.hawkular.metrics.model.MetricType.GAUGE;
+import static org.hawkular.metrics.model.MetricType.GAUGE_RATE;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.hawkular.metrics.alerter.ConditionEvaluator.QueryFunc;
+import org.hawkular.metrics.model.MetricType;
+import org.hawkular.metrics.model.Percentile;
+import org.hawkular.metrics.model.param.Duration;
+import org.hawkular.metrics.model.param.Tags;
+import org.jboss.logging.Logger;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+public class ConditionExpression {
+
+    /**
+     * Determines whether the eval expression is performed on ALL or EACH of the metrics (either specified
+     * explicitly or via the tag query.  ALL means that the metric data points will be combined, then aggregated, and
+     * the evaluation is performed once (per run of the ConditionExpression).  EACH means that data for each
+     * metric will be aggregated separately and the evaluation is performed once per metric.  Note that if multiple
+     * queries are involved in the eval expression that only metrics common to each query will be evaluated.
+     * The default is ALL.
+     */
+    public enum EvalType {
+        ALL, EACH
+    }
+
+    private static final Logger log = Logger.getLogger(ConditionExpression.class);
+
+    /** 10mn */
+    public static final String DEFAULT_FREQUENCY = "10mn";
```

Comment:
"mn" means "minutes" here ? Why don't use just "m" ? In other DSL, the some-sort-standard prefixes are d/h/m/s etc.